### PR TITLE
chore: convert cli tests to md

### PR DESCRIPTION
**Pull request by ssddOnTop**
related to: #2646

Description: This PR converts cli tests to md based tests. This will help to set predefined instructions and add mocks. (for example, setting Env variables).

(ext of #2657)

### CODE REVIEW

File: tests/cli/fixtures/generator/gen_deezer.md [hunk 1]
```diff
@@ -63,3 +64,4 @@
     "query": "Query"
   }
 }
+```
```
Comment (tusharmath) [inline tests/cli/fixtures/generator/gen_deezer.md:67]: No need to wrap with `json`

File: src/cli/generator/config.rs [hunk 2]
```diff
@@ -25,6 +25,8 @@ pub struct Config<Status = UnResolved> {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub preset: Option<PresetConfig>,
     pub schema: Schema,
+    #[serde(default, skip_serializing_if = "TemplateString::is_empty")]
```
Comment (tusharmath) [inline src/cli/generator/config.rs:28]: unnecessary changes.

Comment (ssddOnTop) [reply src/cli/generator/config.rs:28]: existing tests don't even work rn.. the config in tests is either outdated or we need to add the `default` tag on `secret`

**Comment by ssddOnTop** [comment]
closing in favor of #2755
